src/Sentry.Unity.Native/SentryNative.cs

```diff
@@ -24,24 +27,25 @@ public static void Configure(SentryUnityOptions options)
                 };
                 options.ScopeObserver = new NativeScopeObserver(options);
                 options.EnableScopeSync = true;
-                // options.CrashedLastRun = () =>
-                // {
-                //     var crashedLastRun = SentryJava.CrashedLastRun();
-                //     if (crashedLastRun is null)
-                //     {
-                //         // Could happen if the Android SDK wasn't initialized before the .NET layer.
-                //         options.DiagnosticLogger?
-                //             .LogWarning("Unclear from the native SDK if the previous run was a crash. Assuming it was not.");
-                //         crashedLastRun = false;
-                //     }
-                //     else
-                //     {
-                //         options.DiagnosticLogger?
-                //             .LogDebug("Native Android SDK reported: 'crashedLastRun': '{0}'", crashedLastRun);
-                //     }
 
-                //     return crashedLastRun.Value;
-                // };
+                // Note: we must actually call the function now and on every other call use the value we get here.
+                // Additionally, we cannot call this multiple times for the same directory, because the result changes
+                // on subsequent runs. Therefore, we cache the value during the whole runtime of the application.
+                var cacheDirectory = SentryNativeBridge.GetCacheDirectory(options);
+                bool crashedLastRun = false;
+                lock (perDirectoryCrashInfo)
+                {
+                    if (!perDirectoryCrashInfo.TryGetValue(cacheDirectory, out crashedLastRun))
+                    {
+                        crashedLastRun = SentryNativeBridge.HandleCrashedLastRun(options);
+                        perDirectoryCrashInfo.Add(cacheDirectory, crashedLastRun);
+
+                        options.DiagnosticLogger?
+                            .LogDebug("Native SDK reported: 'crashedLastRun': '{0}'", crashedLastRun);
+                    }
+                }
+                options.CrashedLastRun = () => crashedLastRun;
```

Comment:
```suggestion
                // In the event the SDK is re-initialized with a different path on disk, we need to track which ones were already read
                // Similarly we need to cache the value of each call since a subsequent call would return a different value
                // as the file used on disk to mark it as crashed is deleted after we read it.
                lock (_perDirectoryCrashInfo)
                {
                    if (!_perDirectoryCrashInfo.TryGetValue(cacheDirectory, out crashedLastRun))
                    {
                        crashedLastRun = SentryNativeBridge.HandleCrashedLastRun(options);
                        _perDirectoryCrashInfo.Add(cacheDirectory, crashedLastRun);

                        options.DiagnosticLogger?
                            .LogDebug("Native SDK reported: 'crashedLastRun': '{0}'", crashedLastRun);
                    }
                }
                options.CrashedLastRun = () => crashedLastRun;
```